exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,1865 @@
+	if er != nil {
+		return nil, er
+	}
+	var cancelOrderResponse WSPlaceOrderResponse
+	er = json.Unmarshal(respData, &cancelOrderResponse)
+	if er != nil {
+		return nil, er
+	}
+	if cancelOrderResponse.Code != "1" || strings.EqualFold(cancelOrderResponse.Code, "60013") {
+		if cancelOrderResponse.Msg == "" {
+			return nil, errNoValidResponseFromServer
+		}
+		return nil, errors.New(cancelOrderResponse.Msg)
+	} else if len(cancelOrderResponse.Data) == 0 {
+		if cancelOrderResponse.Msg == "" {
+			return nil, errNoValidResponseFromServer
+		}
+		return nil, errors.New(cancelOrderResponse.Msg)
+	}
+	return &(cancelOrderResponse.Data[0]), nil
+}
+
+// WsCancleMultipleOrder cancel multiple order through the websocket channel.
+func (ok *Okx) WsCancleMultipleOrder(args []CancelOrderRequestParam) ([]PlaceOrderResponse, error) {
+	for x := range args {
+		arg := args[x]
+		if arg.InstrumentID == "" {
+			return nil, errMissingInstrumentID
+		}
+		if arg.OrderID == "" && arg.ClientSupplierOrderID == "" {
+			return nil, fmt.Errorf("either order id or client supplier id is required")
+		}
+	}
+	randomID := common.GenerateRandomString(4, common.NumberCharacters)
+	input := WsCancelOrderInput{
+		ID:        randomID,
+		Arguments: args,
+		Operation: "batch-cancel-orders",
+	}
+	respData, er := ok.Websocket.Conn.SendMessageReturnResponse(randomID, input)
+	if er != nil {
+		return nil, er
+	}
+	var cancelOrderResponse WSPlaceOrderResponse
+	er = json.Unmarshal(respData, &cancelOrderResponse)
+	if er != nil {
+		return nil, er
+	}
+	if cancelOrderResponse.Code != "1" ||
+		strings.EqualFold(cancelOrderResponse.Code, "60013") {
+		if cancelOrderResponse.Msg == "" {
+			return nil, errNoValidResponseFromServer
+		}
+		return nil, errors.New(cancelOrderResponse.Msg)
+	}
+	return cancelOrderResponse.Data, nil
+}
+
+// WsAmendOrder method to amend trade order using a request thought the websocket channel.
+func (ok *Okx) WsAmendOrder(arg *AmendOrderRequestParams) (*AmendOrderResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	if arg.ClientSuppliedOrderID == "" && arg.OrderID == "" {
+		return nil, errMissingClientOrderIDOrOrderID
+	}
+	if arg.NewQuantity <= 0 && arg.NewPrice <= 0 {
+		return nil, errMissingNewSizeOrPriceInformation
+	}
+	randomID := common.GenerateRandomString(4, common.NumberCharacters)
+	input := WsAmendOrderInput{
+		ID:        randomID,
+		Operation: "amend-order",
+		Arguments: []AmendOrderRequestParams{*arg},
+	}
+	respData, er := ok.Websocket.Conn.SendMessageReturnResponse("amend-order", input)
+	if er != nil {
+		return nil, er
+	}
+	var amendOrderResponse WsAmendOrderResponse
+	er = json.Unmarshal(respData, &amendOrderResponse)
+	if er != nil {
+		return nil, er
+	}
+
+	if amendOrderResponse.Code != "0" ||
+		amendOrderResponse.Code == "1" ||
+		amendOrderResponse.Code == "60013" {
```

Comment:
The scenarios `"1"` and `"60013"` technically meet the first if check of `!= "0"`. If this `if` statement is setup correctly, then you can just remove the `1`  and `60013` scenarios

Reply:
This message is coming through the WebSocket connection, not the REST HTTP Calls.
So, since the data comes as a raw JSON message, I have to handle this message specifically.
Let me know your suggestion and I will update the code accordingly

Reply:
What I'm saying is, is that this:
```go
if amendOrderResponse.Code != "0" ||
		amendOrderResponse.Code == "1" ||
		amendOrderResponse.Code == "60013" {
```
is the same as:
```go
if amendOrderResponse.Code != "0" {
```
and then that's all you need unless you're looking for specific errors, in which case, please revise your `if` statement

Reply:
Ohh my bad